net/HttpRequest.hpp

```diff
@@ -1380,6 +1380,11 @@ class Session final : public ProtocolHandlerInterface
         }
     }
 
+    net::asyncConnectResult connectionResult()
```

Comment:
Nit: a followup PR could rename this to Async, our types always start with an uppercase letter.